benchmark.py

```diff
@@ -464,23 +498,65 @@ def run_power_test(query_root, data_dir, result, host, port, db_name, user, pass
     try:
         conn = PGDB(host, port, db_name, user, password)
         result.startTimer()
-        if refresh_func1(conn, data_dir, 1):
+        if refresh_func1(conn, data_dir, 0):
             return 1
-        rf1_t = result.stopTimer()
-        result.startTimer()
-        if run_query_stream(conn, query_root, 1):
+        result.setMetric("refresh_func1_stream_%s" % str(0), result.stopTimer())
+        if run_query_stream(conn, query_root, 0, result):
             return 1
-        qs_t = result.stopTimer()
         result.startTimer()
-        if refresh_func2(conn, data_dir, 1):
+        if refresh_func2(conn, data_dir, 0):
             return 1
-        rf2_t = result.stopTimer()
-        result.setSeqTime(rf1_t, qs_t, rf2_t)
+        result.setMetric("refresh_func2_stream_%s" % str(0), result.stopTimer())
     except Exception as e:
         print("unable to run power test. DB connection failed. %s" % e)
         return 1
 
 
+def run_throughput_inner(query_root, data_dir, host, port, db_name, user,
+                            password, stream_num, q):
+    try:
+        conn = PGDB(host, port, db_name, user, password)
+        result = Result("ThroughputQStream_%s" % stream_num)
+        if run_query_stream(conn, query_root, stream_num, result):
+            print("unable to finish query stream %s" % stream_num)
+            exit(1)
+        q.put(result)
+    except Exception as e:
+        print("unable to connect to DB for query stream %s" % stream_num)
+        exit(1)
+
+
+def run_throughput_test(query_root, data_dir, result, host, port, db_name, user, password,
+                        num_streams, q):
+    try:
+        conn = PGDB(host, port, db_name, user, password)
+        result = Result("ThroughputRStream")
+        processes = []
+        for i in range(num_streams):
+            p = Process(target=run_throughput_inner,
+                        args=(query_root, data_dir, host, port, db_name, user, password,
+                                i+1, q))
+            processes.append(p)
+            p.start()
+        for i in range(num_streams):
+            result.startTimer()
+            if refresh_func1(conn, data_dir, i+1):
```

Comment:
According to my understanding from the specification of the TPCH, refresh functions are run in a separate stream but they themselves are run serially. Since we launch separate processes for all query streams I use the master process for the refresh-stream since it is free anyway.